sm/plugins/tedge_docker_plugin/tedge_docker_plugin.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+set -e
+
+COMMAND="$1"
+IMAGE_NAME="$2"
+IMAGE_TAG="$3"
+
+case "$COMMAND" in
+    prepare)
+        # nothing to do here
+        ;;
+    list)
+        sudo docker image list --format '{"name":"{{.Repository}}","version":"{{.Tag}}"}'
```

Comment:
`sudo` doesn't need to be called by the plugin. The sm-agent will call the script with `sudo`.  

That said, running a script under `sudo` might create security issues and imposes strict constraints on the script. See https://unix.stackexchange.com/questions/295422/how-to-secure-a-sudo-powered-script